test/functional/transactions_tests.js

```diff
@@ -205,6 +205,28 @@ describe('Transactions', function() {
       }
     });
   });
+
+  describe('startTransaction', function() {
+    let session, sessionPool;
+    beforeEach(() => {
+      const topology = new core.Mongos();
+      sessionPool = new sessions.ServerSessionPool(topology);
+      session = new sessions.ClientSession(topology, sessionPool);
+    });
+
+    afterEach(() => {
+      sessionPool.endAllPooledSessions();
+    });
+
+    it('should error if transactions are not supported', {
```

Comment:
I like how this uses the core types to make it  a smaller test!

Would it be significantly more difficult to test this with the unified topology?